#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE_NAME = "squirrel_pointcloud_filter"

gen = ParameterGenerator()
gen.add("global_frame_id", str_t, 0, "The reference frame of the world", "/map")
gen.add("update_rate_hz", double_t, 0, "Spinner frequency", 10., 0.0, 100.0)
gen.add("nanfree", bool_t, 0, "No NaN in the input cloud?", True)
gen.add("do_voxel_filter", bool_t, 0, "Downsampling the pointcloud using a voxel fitler", True)
gen.add("do_ground_segmentation", bool_t, 0, "Segment the cloud into ground/non-ground", False)
gen.add("ground_pcls_voxelized", bool_t, 0, "Use voxelized pointcloud for ground filtering", False)
gen.add("ground_threshold", double_t, 0, "", 0.05, 0.0, 1.0)		   
gen.add("resolution_x", double_t, 0, "Voxel filtering resolution (x-coordinate)", 0.05, 0.0, 1.0)
gen.add("resolution_y", double_t, 0, "Voxel filtering resolution (y-coordinate)", 0.05, 0.0, 1.0)
gen.add("resolution_z", double_t, 0, "Voxel filtering resolution (z-coordinate)", 0.05, 0.0, 1.0)

exit(gen.generate(PACKAGE_NAME, "squirrel_pointcloud_filter", "PointCloudFilter"))